code/app/src/main/java/com/example/bread/view/AddMoodEventActivity.java

```diff
@@ -0,0 +1,89 @@
+package com.example.bread.view;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+import com.example.bread.R;
+import com.example.bread.model.MoodEvent;
+import com.example.bread.repository.MoodEventRepository;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class AddMoodEventActivity extends AppCompatActivity {
+    private Spinner emotionalStateSpinner, socialSituationSpinner;
+    private EditText reasonEditText;
+    private Button saveButton;
+    private MoodEventRepository moodEventRepository;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_mood_event);
+
+        // Tie UI elements to variables
+        emotionalStateSpinner = findViewById(R.id.emotionalStateSpinner);
+        reasonEditText = findViewById(R.id.reasonEditText);
+        socialSituationSpinner = findViewById(R.id.socialSituationSpinner);
+        saveButton = findViewById(R.id.saveButton);
+
+        // Set up repository
+        moodEventRepository = new MoodEventRepository();
+
+        // Populate emotional state spinner
+        ArrayAdapter<MoodEvent.EmotionalState> moodAdapter = new ArrayAdapter<>(this,
+                android.R.layout.simple_spinner_item, MoodEvent.EmotionalState.values());
+        moodAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        emotionalStateSpinner.setAdapter(moodAdapter);
+
+        // Populate social situation spinner (add a "None" option for optional)
+        ArrayAdapter<MoodEvent.SocialSituation> socialAdapter = new ArrayAdapter<>(this,
+                android.R.layout.simple_spinner_item, MoodEvent.SocialSituation.values());
+        socialAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        socialSituationSpinner.setAdapter(socialAdapter);
+
```

Comment:
We would also need a place to attach an image with the moodEvent. Not required for now but can you add `TODO` and write what all has to be done.